tcms/signals.py

```diff
@@ -207,21 +207,28 @@ def handle_comments_pre_delete(sender, **kwargs):
 
 
 def handle_emails_post_bug_save(sender, instance, created=False, **kwargs):
-    """
-    Send email updates to assignee after they've been
-    assigned a bug on bug creation.
-    """
-    if not created or instance.assignee is None:
+    from tcms.core.helpers.comments import get_comments
+
+    comments = get_comments(instance)
+
+    if instance.assignee is None or comments.last() is None:
         return
```

Comment:
Since we're changing the behavior you can have a bug where multiple parties are commenting which is still not assigned. The condition should be 
```
if not recipients:
    return
```

Reply:
Still valid.

Reply:
Isn't this is sufficient;

![image](https://user-images.githubusercontent.com/8937774/145712249-e006f0a5-7080-4785-a75f-1865943c7009.png)

IMO, assignee still has to be checked to include its e-mail address to the recipients, maybe I couldn't get your point.

Reply:
Change condition from 
```

    if not recipients or comments.last() is None:
        return
```

to 

```
    if not recipients:
        return
```

Reply:
Not checking if there is a last comment will also result multiple e-mails while bug creation, one called by save and another called by `add_comment`, which is the initial description of the Bug. 

I think this is also acceptable as reopen and close duplication, and I will remove the `comment.last()` check according to your comment:

> That said I don't think there is a quick and easy way to resolve the issue of duplicate emails. My vote goes for merging this PR and living with the duplicate emails as a bug for a while.


---------- MESSAGE FOLLOWS ----------       
b'Content-Type: text/plain; charset="utf-8"'
b'MIME-Version: 1.0'
b'Content-Transfer-Encoding: 7bit'
b'Subject: [Kiwi-TCMS] Bug #33 - Summary of the Bug'
b'From: kiwi@example.com'
b'To: emailaddress1@gmail.com, emailaddress2@gmail.com'
b'Date: Mon, 13 Dec 2021 11:40:48 -0000'
b'Message-ID: '
b' <163939564892.12340.4210047131531209881@hostname>'
b'X-Peer: 127.0.0.1'
b''
b''
b'Bug 33 has been updated.'
b''
b'Link: http://127.0.0.1:8000/bugs/33/'
b''
b'Summary: Summary of the Bug'
b'Created at: Dec. 13, 2021, 2:40 p.m.'
b'Reporter: cmb'
b'Assignee: cmb'
b'Product: Sample Product'
b'Version: Version 1.0'
b'Build: INC1'
b'Last comment: '
------------ END MESSAGE ------------
---------- MESSAGE FOLLOWS ----------
b'Content-Type: text/plain; charset="utf-8"'
b'MIME-Version: 1.0'
b'Content-Transfer-Encoding: 7bit'
b'Subject: [Kiwi-TCMS] Bug #33 - Summary of the Bug'
b'From: kiwi@example.com'
b'To: emailaddress1@gmail.com, emailaddress2@gmail.com'
b'Date: Mon, 13 Dec 2021 11:40:49 -0000'
b'Message-ID: '
b' <163939564904.12340.16028265939125036183@hostname>'
b'X-Peer: 127.0.0.1'
b''
b''
b'Bug 33 has been updated.'
b''
b'Link: http://127.0.0.1:8000/bugs/33/'
b''
b'Summary: Summary of the Bug'
b'Created at: Dec. 13, 2021, 2:40 p.m.'
b'Reporter: cmb'
b'Assignee: cmb'
b'Product: Sample Product'
b'Version: Version 1.0'
b'Build: INC1'
b'Last comment: '
b'Description of problem:'
b''
b''
b'How often reproducible:'
b''
b''
b'Steps to Reproduce:'
b'1.'
b'2.'
b'3.'
b''
b'Actual results:'
b''
b''
b'Expected results:'
b''
b''
b'Additional info:'
------------ END MESSAGE ------------